lib/hexpm/web/session.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Hexpm.Web.Session do
+  import Ecto.Query
+
+  alias Hexpm.Accounts.Session
+  alias Hexpm.Repo
+
+  @behaviour Plug.Session.Store
+  @fake_token <<0::96*8>>
+
+  def init(_opts) do
+    :ok
+  end
+
+  def get(_conn, cookie, _opts) do
+    {id, token} = parse_cookie(cookie)
+    session = Repo.get(Session, id)
+    if session && Plug.Crypto.secure_compare(token, session.token) do
+      {id, session.data}
```

Comment:
You can make the id be `{id, token}`. This way you don't need to use update_all with returning on  `put`. And there is no need to generate a fake token.